### ISSUE TITLE
HashLink is not included with standard Haxe distribution

### ISSUE THREAD

**Issue by TheSpydog** [HaxeFoundation/haxe#9546]
As far as I can tell, there is no option to install HashLink during the default Haxe installation process. This is confusing to me, since the messaging on the Haxe website communicates that HL is effectively the successor to Neko and should be preferred where possible, and yet it remains a separate download. Relegating HL to a standalone binary (and its corresponding haxelib package) makes it seem less "officially supported" than Neko or the other built-in targets, and increases the possibility of accidentally using an incompatible version of HL after upgrading Haxe since they're not packaged together.

The situation is made worse since the current UX for installing and using HL isn't great IMO, especially on Windows since it requires manual extraction and modifying the system path. It's also not entirely clear at first blush why you can't just `haxelib run hashlink <your-file.hl>` without installing a separate binary... maybe if the binary was installed via the hashlink haxelib (like hxcpp) the non-inclusion in the default installation wouldn't be so jarring.

**Comment by RealyUniqueName** [comment]
I don't think we will add hashlink to the Haxe installer. Not all Haxe users target Hashlink.
But perhaps we need a separate "All-in-one" package.

**Comment by back2dos** [comment]
Well, to be fair we do include neko, so expecting to find HL in there instead is not *that* unreasonable.

**Comment by RealyUniqueName** [comment]
Neko is required to run `haxelib run`.

**Comment by TheSpydog** [comment]
Neko is listed in the install process as an optional component. This is what I was envisioning for HashLink as well. Users who want it can enable it, and those who don't care about it don't have to install it.

![install](https://user-images.githubusercontent.com/8942401/84080927-02d16300-a9ab-11ea-917a-f43cd002d79b.PNG)

**Comment by RealyUniqueName** [comment]
Ah, indeed. That might be an option.

**Comment by TheSpydog** [comment]
I'm willing to contribute this feature, but I would appreciate some guidance since I've never worked with the internals of the Haxe installer. From browsing around the repo it looks like all the relevant code is housed inside the `extra` directory, is that correct?

Also, should HL be (optionally) included in _all_ the supported installation methods for all platforms? For example, I see that there's a `haxe.nuspec` file for Chocolatey that [depends on Neko](https://github.com/HaxeFoundation/haxe/blob/development/extra/choco/haxe.nuspec#L25). Should HL also be added to Chocolatey and added as a dependency for consistency?

Thanks!

**Comment by RealyUniqueName** [comment]
@andyli any hints?

**Comment by russellballestrini** [comment]
To compound confusion even further (as a two day old novice to the `haxe` ecosystem), I was able to install `HashLink` using `haxelib install hashlink` but the version which is installed appears out-of-date (`hashlink: [0.1.0]`) and also at least on my OS distribution (Fedora), it doesn't install the `hl` command to my path.

```
haxelib list | grep hash
```

What is the preferred method of installation for HashLink?

I've got `neko` and `html5` and `linux` targets working for my system but I started poking at `heaps` instead of `haxeflixel` and this is where I got hung up.
